app/src/main/java/org/mozilla/fenix/RunBlockingCounter.kt

```diff
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This class implements the alternative ways to invoke runBlocking with some
+// monitoring by wrapping the raw methods. This lint check tells us not to use the raw
+// methods so we suppress the check.
+@file:Suppress("MozillaRunblockingCheck")
+
+package org.mozilla.fenix
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.runBlocking
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * Counts the number of runBlocking calls made
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+class RunBlockingCounter {
+    companion object {
+        var runBlockingCount = 0L
+    }
+}
+
+/**
+ * Wrapper around `runBlocking`
+ */
+fun <T> runBlockingIncrement(
+    context: CoroutineContext = EmptyCoroutineContext,
```

Comment:
I think you should use null to signify the absence of a value passed in. `EmptyCoroutineContext` may be a reasonable parameter to pass in now or in the future so we shouldn't use it as the optional argument state.

Reply:
That's a fair point. I just tried to keep our function signature as closed as possible to the original `runBlocking` which takes an `EmptyCoroutineContext`